### PR TITLE
Update to xenial

### PR DESCRIPTION
**Pull request by probonopd**

### CODE REVIEW

File: source/reference/best-practices.rst [hunk 1]
```diff
@@ -87,7 +87,7 @@ If you run into errors like this::
...
 
-When producing AppImages for the Subsurface project, I have had very good results by using **CentOS 6**. This distribution is not too recent (current major CentOS version minus 1) while there are still the most recent Qt and modern compilers for it in the EPEL_ and devtools-2_ (the community equivalent of the Red Hat Developer Toolset 2) repositories. When using it for compilation, I found the resulting binaries to run on a wide variety of systems, including **debian oldstable** (wheezy).
+When producing AppImages for the Subsurface project, we have had very good results by using **CentOS 6**. This distribution is not too recent (current major CentOS version minus 1) while there are still the most recent Qt and modern compilers for it in the EPEL_ and devtools-2_ (the community equivalent of the Red Hat Developer Toolset 2) repositories. When using it for compilation, I found the resulting binaries to run on a wide variety of systems, including **debian oldstable** (wheezy).
```
Comment (TheAssassin) [inline source/reference/best-practices.rst]: `wheezy` isn't oldstable any more. I have a policy for my own texts to ideally use aliases (like "oldest still-supported LTS" or, like here, "oldstable"), combined with a link to the latest information.

File: source/reference/best-practices.rst [hunk 2]
```diff
@@ -113,4 +113,4 @@ libstdc++.so.6
 --------------
 
 .. note::
-	**As a general rule of thumb, please use no libstdc++.so.6 newer than the one that comes with the oldest distribution that you still want to support, i.e., the oldest still-supported LTS version** (at the time of this writing, Ubuntu 14.04).
+	**As a general rule of thumb, please use no libstdc++.so.6 newer than the one that comes with the oldest distribution that you still want to support, i.e., the oldest still-supported LTS version** (at the time of this writing, Ubuntu 16.04).
```
Comment (TheAssassin) [inline source/reference/best-practices.rst]: Maintaining any sort of version number is a bad idea. Here, combined with the lack of a date when this was last updated ("at the time of writing", well, when was that?), the reader is confused. See above.

Two options: add a date when this was last updated, or, better, do not put in any hardcoded version number but place a link to up to date information.

**Comment by probonopd** [comment]
Yes, maybe just use links to Debian and Ubuntu where one can see which version is currently the oldest still-supported release.

**Comment by TheAssassin** [comment]
I improved the style of these paragraphs a bit, updated the data on CentOS, and inserted some snippets to make the text more future-proof. Should be good to merge now.

**Pull request merged**